### ISSUE TITLE
Implement occurrence download

### ISSUE THREAD

**Issue by timrobertson100** [gbif/portal16#63]
- [x] Download button on occurrence page pops up form / open new page
- [x] If required user is prompted to log in
- [x] Let user select download format
- [x]  Show user summary of issues, suggesting e.g. `hasGeospatialIssue=false` etc

Blocked by #62

**Comment by kbraak** [comment]
I'd like to propose trying to dissuade users from downloading the entire index. If we detect they are about the download the entire index, ask them to confirm this is intended and that they understand a suite of facets and filters are available for use. 

I'd also like to propose adding a disclaimer making the user aware that:

- GBIF assigns the download a DOI with the intention it makes citing the download easier
- GBIF assumes the responsibility of ensuring this DOI resolves indefinitely
- We encourage the user to either 
  - associate the download with a journal article or to indicate their intended purpose for the download  (see issue #64)
  - delete this download if they know it will be unused, is a duplicate, etc (see issue #64)
- We explain that the download may be deleted/cleaned up after a specified time

**Comment by DimEvil** [comment]
In some repositories they give the downloader the opportunity to share the reason of the download, in a simple text box. Wouldn't it be nice to have this for GBIF downloads. Not mandatory, but just the opportunity to share the 'story-reason' of downloading data. This might give us more overall insight on downloading events.

**Comment by MortenHofft** [comment]
Currently the user has to enter username and password every time as there isn't any login. But it is possible to download and there is added a basic report of issues in the data and a few other warnings (living specimens and fossils).
